pints/_log_pdfs.py

```diff
@@ -115,6 +115,161 @@ def sample(self, n=1):
         raise NotImplementedError
 
 
+class PooledLogPDF(LogPDF):
+    r"""
+    Combines :math:`m` :class:`LogPDFs<pints.LogPDF>`, each with :math:`n`
```

Comment:
Wonder if having n_1, n_2, ..., n_m parameters for the m models with k parameters of that being in common and pooled, where n_i does not necessary equal to n_j, would be useful? Unlike hierarchical models, I suppose there's no rule or such like that the parameters in the m models apart from the k parameters have to be the same?

Reply:
I was wondering the same at first. What speaks against doing that are mainly two things: 1. I couldn't really come up with a biologically motivated problem that would require pooling parameters across different ODE models, but we could maybe find such a problem; 2. The interface for pooling across models with varying parameter spaces would have to be much more complicated, because you would have to specify which parameter (i.e position) across all individuals is going to be published, and you would have to do that for every parameter that you want to pool. 

So I would suggest that we leave this for now, unless someone has a really convincing use case. What do you think?

Reply:
Hmm... I don't know, what kind of annoys me is that it looks like "this method is very general, but not that general"...

Reply:
I'd say it's as general as it can be for biologically relevant problems, and the docstring specifies that all likelihoods have n parameters. 

I personally don't think that the gain in generality for allowing likelihoods with varying parameter spaces outweighs the benefits of a simple api for the use case of having likelihoods with the same parameter space. 

Anyway I'm happy to discuss this in the next meeting.

Reply:
We decided to keep this simpler, so what it is now will do!